### ISSUE TITLE
[Bug]: is missing in props validation  when wrapped  by Memo 

### ISSUE THREAD

**Issue by vaynevayne** [jsx-eslint/eslint-plugin-react#3550]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

```jsx

type NavbarSegmentedProps = {
  authRouters: MetaMenuAuthRouteObject[]
  opened: boolean
  setOpened: (opened: boolean) => void
} & Omit<NavbarProps, 'children'>

export const NavbarSegmented: FC<NavbarSegmentedProps> = memo(
  ({ authRouters, opened, setOpened, ...props }) => {
    
  }
)
```

  24:6   error    'authRouters' is missing in props validation      react/prop-types
  24:19  error    'opened' is missing in props validation           react/prop-types
  24:27  error    'setOpened' is missing in props validation        react/prop-types
  53:46  error    'authRouters.map' is missing in props validation 

### Expected Behavior

it is ok

### eslint-plugin-react version

7.32.2

### eslint version

8.30.0

### node version

v16.16.0

**Comment by vaynevayne** [comment]
.eslintrc.json

```jsx
{
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:react-hooks/recommended"
  ],
  "overrides": [],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module"
  },
  "plugins": [
    "react",
    "react-refresh",
    "@typescript-eslint",
    "@tanstack/query"
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "rules": {
    "react/react-in-jsx-scope": "off",
    "react/self-closing-comp": ["error"],
    "quotes": ["error", "single"],
    "semi": ["error", "never"],
    "react/display-name": ["off"],
    "@typescript-eslint/no-explicit-any": "off",
    "@tanstack/query/exhaustive-deps": "error",
    "react-refresh/only-export-components": "warn",
    "curly": ["error", "multi"]
  }
}

```

**Comment by thep0y** [comment]
same here.

**Comment by Charlygraphy23** [comment]
Have you tried like this- 

```js

type NavbarSegmentedProps = {
  authRouters: MetaMenuAuthRouteObject[]
  opened: boolean
  setOpened: (opened: boolean) => void
} & Omit<NavbarProps, 'children'>

 const NavbarSegmented: FC<NavbarSegmentedProps> =  ({ authRouters, opened, setOpened, ...props }) => {}
  
  export default memo(NavbarSegmented)
```

**Comment by vaynevayne** [comment]
@Charlygraphy23  second way is ok, but cause new warning  with `eslint-plugin-react-refresh`

https://github.com/ArnaudBarre/eslint-plugin-react-refresh/issues/6

<img width="967" alt="image" src="https://user-images.githubusercontent.com/124666577/230289763-d272c075-1037-4ca1-ac25-14c4fe310701.png">

**Comment by ljharb** [comment]
So name it - components shouldn't be arrow functions.

**Comment by vaynevayne** [comment]
it works ,  thanks
